crates/uv-build-backend/src/lib.rs

```diff
@@ -32,6 +32,8 @@ use crate::settings::ModuleName;
 pub enum Error {
     #[error(transparent)]
     Io(#[from] io::Error),
+    #[error("Failed to persist temporary file to {}", _0.user_display())]
+    Persist(PathBuf, #[source] tempfile::PersistError),
```

Comment:
I think having a custom error here is probably a good idea, but just a small request:

`tempfile::PersistError` contains both the `io::Error` and the `NamedTempFile` we failed to persist. We're not making use of the whole `NamedTempFile` and don't want to keep it around unnecessarily while we do error handling, we just need the `io::Error` inside.